actionpack/lib/action_dispatch/middleware/stack.rb

```diff
@@ -192,7 +192,7 @@ def index_of(index)
         end
 
         middlewares.index do |m|
-          m.klass == index || (@rack_runtime_deprecated && m.klass == FakeRuntime && index == Rack::Runtime)
+          m.name == klass.name || (@rack_runtime_deprecated && m.klass == FakeRuntime && klass == Rack::Runtime)
```

Comment:
The only functional change in this method is here (`m.klass == index` becomes `m.name == klass.name`). But the variable being called `index` was mega confusing when we know it's not an integer at this stage so I renamed it too.